_monitoring-plugins/ad/api.md

```diff
@@ -368,297 +380,1546 @@ POST _plugins/_anomaly_detection/detectors/<detectorId>/_preview
     },
     "detection_interval": {
       "period": {
-        "interval": 10,
-        "unit": "MINUTES"
+        "interval": 1,
+        "unit": "Minutes"
       }
     },
     "window_delay": {
       "period": {
         "interval": 1,
-        "unit": "MINUTES"
+        "unit": "Minutes"
       }
     },
+    "shingle_size": 8,
     "schema_version": 0,
     "feature_attributes": [
       {
-        "feature_id": "xxokEnIBcpeWMD987A1X",
-        "feature_name": "total_order",
+        "feature_id": "U0HKTXwBwf_U8gjUXY2m",
+        "feature_name": "test",
         "feature_enabled": true,
         "aggregation_query": {
-          "total_order": {
+          "test": {
             "sum": {
               "field": "value"
             }
           }
         }
       }
     ],
-    "last_update_time": 1589442309241
+    "last_update_time": 1633392680364,
+    "user": {
+      "name": "admin",
+      "backend_roles": [
+        "admin"
+      ],
+      "roles": [
+        "own_index",
+        "all_access"
+      ],
+      "custom_attribute_names": [],
+      "user_requested_tenant": "__user__"
+    },
+    "detector_type": "SINGLE_ENTITY"
   }
 }
 ```
 
-If you specify a category field, each result is associated with an entity:
-
-#### Sample response
-
-```json
-{
-  "anomaly_result": [
-    {
-      "detector_id": "4CIGoHUBTpMGN-4KzBQg",
-      "data_start_time": 1604277960000,
-      "data_end_time": 1604278020000,
-      "schema_version": 0,
-      "anomaly_grade": 0,
-      "confidence": 0.99
-    }
-  ],
-  "entity": [
-    {
-      "name": "host",
-      "value": "i-00f28ec1eb8997686"
-    }
-  ]
-},
-{
-  "detector_id": "4CIGoHUBTpMGN-4KzBQg",
-  "data_start_time": 1604278020000,
-  "data_end_time": 1604278080000,
-  "schema_version": 0,
-  "feature_data": [
-    {
-      "feature_id": "0Kld3HUBhpHMyt2e_UHn",
-      "feature_name": "latency_max",
-      "data": -17
-    }
-  ],
-  "anomaly_grade": 0,
-  "confidence": 0.99,
-  "entity": [
-    {
-      "name": "host",
-      "value": "i-00f28ec1eb8997686"
-    }
-  ]
-}
-...
-
-```
-
----
-
-## Start detector job
-Introduced 1.0
-{: .label .label-purple }
-
-Starts a real-time or historical anomaly detector job.
-
-
-#### Request
-
-```json
-POST _plugins/_anomaly_detection/detectors/<detectorId>/_start
-```
-
-#### Sample response
-
-```json
-{
-  "_id" : "m4ccEnIBTXsGi3mvMt9p",
-  "_version" : 1,
-  "_seq_no" : 6,
-  "_primary_term" : 1
-}
-```
-
-
----
-
-## Stop detector job
-Introduced 1.0
-{: .label .label-purple }
-
-Stops a real-time or historical anomaly detector job.
+Use `job=true` to get real-time analysis task information.
```

Comment:
How about change "task" to "job" here?
In AD,  "job" is something scheduled and will run periodically, and we only have such scheduled thing for realtime anomaly detection. For historical analysis, it just one-time run, not scheduled and won't run periodically. 

When user start realtime detection for a detector, AD plugin will create one "job" or update existing job as enabled if job exists. Add AD plugin will create a new realtime task when user start/restart realtime detection for a detector, and that realtime task will record runtime information like detector configuration snapshot, realtime job state like initializing/running/stopped, init progress etc. One detector can only have one realtime job (job id is the same with detector id), but can have multiple realtime tasks (each restart of realtime job will create a new realtime task and the realtime task number is limited by cluster setting "plugins.anomaly_detection.max_old_ad_task_docs_per_detector")

For historical analysis, we have no "job". When user starts/rerun historical analysis for a detector, AD plugin will create a new historical batch task which tracks the historical analysis runtime information like state, coordinating/worker node, task progress etc. And the historical task number will be limited by cluster setting "plugins.anomaly_detection.max_old_ad_task_docs_per_detector" as well.